cmd/strelaysrv/etc/linux-systemd/system/strelaysrv.service

```diff
@@ -0,0 +1,22 @@
+[Unit]
+Description=Syncthing Relay Server
+After=network.target
+Documentation=man:strelaysrv(1)
+
+[Service]
+WorkingDirectory=/var/lib/syncthing-relaysrv
+EnvironmentFile=/etc/default/syncthing-relaysrv
+ExecStart=/usr/bin/strelaysrv $STRELAYSRV
+Nice=15
+
+# Hardening
+User=syncthing-relaysrv
+Group=syncthing
+PrivateTmp=yes
+ReadOnlyPaths=/usr
```

Comment:
Yes I wanted to ask you if `ReadOnly=/usr` means that `/` is inaccessible or else how to block `/`, because `/usr` and `/var/lib/syncthing-relaysrv` should be sufficient.
Also I gave you access so feel free to add that other things.

Reply:
You can completely hide a directory tree with `InaccessiblePaths=path`, however I would NOT recommend it, because it will break everything left and right.
For example the libc reads locale form `/etc/locale.conf`, DNS uses `/etc/resolv.conf`, and probably many other things I don't even know about (try strace on a random program that does non trivial things, and be amazed at the complexity of a modern Linux system).

I most case (and when it is possible) it is safe to set the mount namespace with everything read only except the small subset of path we can write to (thankfully with the relay is is very simple). The other options on paths I mentioned above are worthy addition because they hide common subtrees where permissions might not be restrictive enough: 
* `PrivateDevices`:  because writing to a disk device ie. `/dev/sdx` means we can do anything
* `ProtectHome`: because no automated program should go in `/home`, and some careless users might have `~/.ssh/id_rsa` world readable
* `PrivateTmp`: because it is shared by all programs when in most case there is no need to

Reply:
Then do that and the `ProtectSystem=full` thing

Reply:
I've just basic knowledge about that and never played with all that restriction stuff, but relied on canonical knowing what they are doing.

But I you start, just do it correctly from the beginning.

Reply:
`ProtectSystem=full` is strictly equivalent of:
```
ReadOnlyPaths=/boot /usr /etc
```

`ReadOnlyPaths=/` is much more defensive and sets everything read only, so this is what I would use. In theory it is more likely to break stuff than `ProtectSystem=full`, but:
* a relay only does... relay and unlike syncthing does not write to shared folders
* it runs fine on my Ubuntu server

Reply:
I have a patch ready for the hardening, I'm testing on my server for a while before pushing.

Reply:
Do what you think works best, we'll have to test the final deb any at the end